### PR TITLE
Feature/801 fleet mode

### LINKED ISSUE #801: Create "Fleet Mode" for App users with many many Hotspots
Reported by Android user on 3.4.1 with 100+ Hotspots in the account, using Samsung Galaxy Tab.

https://explorer.helium.com/accounts/139DUUzVJcR6NcvjkXGJzSpaWLXUjRDvGMEQuHCDkikybzePfn2

### PR DESCRIPTION
**Pull request by matthewcarlreetz**
Closes #801

### CODE REVIEW

File: src/features/hotspots/root/HotspotsScreen.tsx [hunk 1]
```diff
@@ -46,12 +87,8 @@ const HotspotsScreen = () => {
   })
 
   useEffect(() => {
-    // dispatch(fetchHotspotsData()) will trigger an update to hotspots
-    // anytime hotspots update, update rewards and following
-    // Separating these calls allows the UI to respond more quickly
-    dispatch(fetchFollowedHotspotsFromBlock())
-    dispatch(fetchRewards())
-  }, [hotspots, dispatch])
+    dispatch(fetchRewards({ fetchType: fleetModeEnabled ? 'followed' : 'all' }))
```
Comment (allenan) [inline src/features/hotspots/root/HotspotsScreen.tsx:90]: is there some state they could get into where they follow too many hotspots and it bogs down too much to where they can't get in to unfollow them? Should there be an upper limit on followed hotspots? at least until we have the API sorted out?

File: src/store/user/appSlice.ts [hunk 2]
```diff
@@ -49,13 +53,15 @@ export const restoreUser = createAsyncThunk<Restore>(
   'app/restoreUser',
   async () => {
     const vals = await Promise.all([
-      getSecureItem('accountBackedUp'),
```
Comment (allenan) [inline src/store/user/appSlice.ts:52]: you could potentially destructure this like `const [accountBackedUp, requirePin, ...] = await Promise.all([...])` if you don't want to keep track of which index each item is in

File: src/utils/appDataClient.ts [hunk 3]
```diff
@@ -140,6 +140,20 @@ export const getAccount = async (address?: string) => {
...
+  address?: string
+  numDaysBack?: number
+}) => {
+  Logger.breadcrumb('getAccountRewards', breadcrumbOpts)
+  const accountAddress = opts?.address || (await getAddress())
+  if (!accountAddress) return
+
+  const initialDate = new Date()
+  const endDate = new Date()
+  endDate.setDate(initialDate.getDate() - (opts?.numDaysBack || 1))
```
Comment (allenan) [inline src/utils/appDataClient.ts:153]: this works, there are also some handy functions in the `date-fns` package for working with stuff like this

**Review by allenan — APPROVED** [review]
looks good to me

File: src/utils/appDataClient.ts (hunk 3 above)
Comment (matthewcarlreetz) [reply src/utils/appDataClient.ts:153]: yeah, that's what I would normally use, but I copied this from explorer 😄

Comment (matthewcarlreetz) [reply src/utils/appDataClient.ts:153]: I'll use date-fns to be consistent with other date math in the app.

File: src/store/user/appSlice.ts (hunk 2 above)
Comment (matthewcarlreetz) [reply src/store/user/appSlice.ts:52]: yes! That's so much better.

File: src/features/hotspots/root/HotspotsScreen.tsx (hunk 1 above)
Comment (matthewcarlreetz) [reply src/features/hotspots/root/HotspotsScreen.tsx:90]: Yes, we discussed this at one point and decided not to handle it for the moment. I will raise the question again and can address it in a separate PR if needed.

**Pull request merged**